packages/share_plugin/ios/Classes/SwiftSharePlugin.swift

```diff
@@ -0,0 +1,28 @@
+import Flutter
+import UIKit
+
+public class SwiftSharePlugin: NSObject, FlutterPlugin {
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(name: "share_plugin", binaryMessenger: registrar.messenger())
+    let instance = SwiftSharePlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+         if (call.method == "shareMethod") {
+              let arguments = call.arguments as! [String: Any]
+              let message = arguments["message"] as! String
```

Comment:
throw an exception with message `"It's required to pass the message param to the plugin"`. u can do it via `guard` construction (https://docs.swift.org/swift-book/LanguageGuide/ErrorHandling.html)